src/main/java/uk/gov/justice/digital/delius/service/UserService.java

```diff
@@ -85,6 +83,18 @@ public AccessLimitation accessLimitationOf(final String subject, final OffenderD
                         .build()).collect(toList());
     }
 
+    public Map<String, Optional<UserDetails>> getUserDetailsMap(final Set<String> usernames) {
+        return usernames.stream().collect(Collectors.toMap(username -> username, username -> ldapRepository.getDeliusUser(username).map(user ->
+                UserDetails
+                        .builder()
+                        .roles(user.getRoles().stream().map(role -> UserRole.builder().name(role.getCn()).build()).collect(toList()))
```

Comment:
I think I would be tempted to *not* add the roles, I don't think clients of this will ever need to the roles and are very internal to Delius (Auth needs to but not anyone else)

Reply:
agreed - dropped and pushed up again